### PR TITLE
Docs/rtd staging

### PR DESCRIPTION
**Pull request by ederjc**
**Preview now available at: https://xmc-arduino-staging.readthedocs.io**

### Description
First version of the new created ReadTheDocs Documentation. Also a new Readme file.

### Context
Ready for Review are allready:

- Home
- Installation Instructions
- Supported Boards
- README

### CODE REVIEW

File: docs/installation-instructions.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+.. image:: img/jlink_install.png
+    :width: 700
+
+If you are downloading J-Link for the first time, make sure to check this box: Install USB Driver for J-Link.
+
+.. image:: img/J-Link_Installer_options_page.png
+    :width: 650
+
+Required tools
```
Comment (LinjingZhang) [inline docs/installation-instructions.rst:22]: Sorry I updated the content here, it needs to be added: https://github.com/Infineon/XMC-for-Arduino?tab=readme-ov-file#required-tools
pip's link was displayed incorrectly

File: docs/hw-platforms.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+The XMC1400 2Go kit provides an easy way to evaluate almost all capabilities of the XMC1400 microcontroller.  
+The kit is powered via USB, interfaces  to other Infineon sensor boards and provides multiple interfaces including a 
+CAN bus. The software development is supported via ModusToolbox™ and the Arduino IDE.  
+To find out more visit the Infineon homepage of the `KIT_XMC14_2GO`_.
+
+Pinout Diagram
+^^^^^^^^^^^^^^
+
+.. image:: img/XMC_4700_RelaxKit_5VShields.png
```
Comment (LinjingZhang) [inline docs/hw-platforms.rst:20]: Wrong pinout for XMC14 2go?

File: docs/hw-platforms.rst [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    :width: 600
+
+
+
+
+
+
+KIT_XMC11_BOOT_001
+-------------------
```
Comment (LinjingZhang) [inline docs/hw-platforms.rst:29]: The naming is a bit weird for me.
The product name doesn't match the name inside the pinout diagram/IDE (code)

**Review by LinjingZhang — COMMENTED** [review]
Thanks to @CaGriesser , we finally have RTD for documentation, very good starting point 😻

File: docs/installation-instructions.rst (hunk 1 above)
Comment (ederjc) [reply docs/installation-instructions.rst]: @CaGriesser

File: docs/hw-platforms.rst (hunk 3 above)
Comment (jaenrig-ifx) [reply docs/hw-platforms.rst]: Yep, the SCREAMING_SNAKE_CASE for the product ordering number also does not feel the best for me. 
image
I would add not the product page title. The second one ?

**Comment by jaenrig-ifx** [comment]
As mentioned in the other PR, i think this info is redundant, and available in the product page.
image

https://www.infineon.com/cms/en/product/evaluation-boards/kit_xmc1400_arduino/

**Comment by jaenrig-ifx** [comment]
We will iterate over this, but very nice to see how the docs are moving to RTD! 📜

**Review by jaenrig-ifx — APPROVED** [review]

**Pull request merged**